### PR TITLE
CBG-4856 do not create merge versions for remote wins/local wins

### PR DESCRIPTION
**Pull request by torcolvin**
CBG-4856 do not create merge versions for remote wins/local wins

- Only create new merge versions if there is a change in body.
- The way the ISGR  tests are modified is a bit messy but out of scope for this change.
- Fix an edge case merging two vv: 	`"10@abc,1@def,3@efg;1@foo"` and   `"2@xyz,8@def,9@efg;1@foo"`, this is the change in `UpdateHistory`

This change and https://github.com/couchbase/couchbase-lite-core/pull/2352 should match behavior

## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGatewayIntegration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGatewayIntegration/92/

### CODE REVIEW

File: db/hybrid_logical_vector.go [hunk 1]
```diff
@@ -866,27 +893,15 @@ func DefaultLWWConflictResolutionType(ctx context.Context, conflict Conflict) (B
...
-	base.DebugfCtx(ctx, base.KeyVV, "resolved conflict for doc %s in favour of local wins, resulting HLV: %v", base.UD(docID), newHLV)
-
+func localWinsConflictResolutionForHLV(ctx context.Context, localHLV, incomingHLV *HybridLogicalVector, docID string) (*HybridLogicalVector, error) {
+	if localHLV == nil {
+		return nil, errors.New("localHLV is nil for localWinsConflictResolutionForHLV")
+	} else if incomingHLV == nil {
+		return nil, errors.New("incomingHLV is nil for localWinsConflictResolutionForHLV")
+	}
+	newHLV := incomingHLV.Copy()
+	newHLV.UpdateWithIncomingHLV(localHLV)
```
[the remark this answers is not in the document]
Comment (gregns1) [reply db/hybrid_logical_vector.go]: This definitely confused me when I saw this yesterday, I had to actually go through the function to understand how its correct. I maybe would include a short comment explaining this.

File: rest/replicatortest/replicator_conflict_test.go [hunk 2]
```diff
@@ -1641,10 +1670,20 @@ func TestActiveReplicatorHLVConflictCustom(t *testing.T) {
 			expectedBody:            `{"source":"merged"}`,
 			remoteBody:              `{"source":"remote"}`,
 			expectedDocPushResolved: true,
-			mergeVersionsExpected:   true,
-			newCVGenerated:          true,
+			hlvResult: db.HybridLogicalVector{
+				SourceID: "NEED TO EXPAND IN TEST",
+				Version:  math.MaxInt64, // # this needs to get expanded in test code to doc.Cas before comparison
```
Comment (gregns1) [inline rest/replicatortest/replicator_conflict_test.go:1679]: `math.MaxUint64` is what's used to signal to macro expand

Comment (gregns1) [reply rest/replicatortest/replicator_conflict_test.go:1679]: Scratch that, I see below this value is assign doc cas

**Review by gregns1 — CHANGES_REQUESTED** [review]
Looks good just comment change + suggestion then ready to go

**Comment by torcolvin** [comment]
I combined the local and remote tests to make it easier for Couchbase Lite folks to follow what is happening, let me know if this is easier or harder to understand by dropping local/remote wins explicit functions.

File: db/change_cache.go [hunk 3]
```diff
@@ -505,6 +505,15 @@ func (c *changeCache) DocChanged(event sgbucket.FeedEvent, docType DocumentType)
 	if len(rawUserXattr) > 0 {
 		collection.revisionCache.RemoveRevOnly(ctx, docID, syncData.GetRevTreeID())
 	}
+	// remove the local doc from the revision cache if the change is a result of a conflict resolution that resulted
+	// in local/remote wins, given the HLV will have been updated but CV not changed
```
Comment (gregns1) [inline db/change_cache.go]:
```suggestion
	// in local wins, given the HLV will have been updated but CV has not changed
```

**Review by gregns1 — APPROVED** [review]
